### ISSUE TITLE
duplicate `domain_of` values in specification of meta model

### ISSUE THREAD

**Issue by vincentvialard** [linkml/linkml#1147]
**Describe the bug**
`domain_of` in meta model contains duplicates

**To reproduce**
1. Go to [Direct LinkML Specification of SchemaDefinition](https://linkml.io/linkml-model/docs/SchemaDefinition/#direct)
2. Click on 'Details'
3. Scroll down to `slot_usage`
4. See error (duplicate 'element')

```
slot_usage:
  name:
    name: name
    description: a unique name for the schema that is both human-readable and consists
      of only characters from the NCName set
    domain_of:
    - element
    - element
    range: ncname
tree_root: true
```
This occurs in other places, for instance [Direct LinkML Specification of SchemaDefinition](https://linkml.io/linkml-model/docs/SlotDefinition/#direct)

**Expected behavior**
Duplicates are not expected. 

*It seems that domains from `slots` and `slot_usage` are appended instead of merged.*

**Comment by turbomam** [comment]
This effects the nmdc-schema too

**Comment by sujaypatil96** [comment]
Unit test for the fix is in PR #1319
